source/opt/const_folding_rules.cpp

```diff
@@ -350,14 +350,63 @@ ConstantFoldingRule FoldFPBinaryOp(BinaryScalarFoldingRule scalar_rule) {
   return [scalar_rule](IRContext* context, Instruction* inst,
                        const std::vector<const analysis::Constant*>& constants)
              -> const analysis::Constant* {
+    analysis::ConstantManager* const_mgr = context->get_constant_mgr();
+    analysis::TypeManager* type_mgr = context->get_type_mgr();
+    const analysis::Type* result_type = type_mgr->GetType(inst->type_id());
+    const analysis::Vector* vector_type = result_type->AsVector();
+
     if (!inst->IsFloatingPointFoldingAllowed()) {
       return nullptr;
     }
-    if (inst->opcode() == SpvOpExtInst) {
-      return FoldFPBinaryOp(scalar_rule, inst->type_id(),
-                            {constants[1], constants[2]}, context);
```

Comment:
This looks like it will disable folding extended instructions.

Reply:
In general, it looks like you changed an older version of this code, and you are undoing the changes that have been done lately.  See https://github.com/KhronosGroup/SPIRV-Tools/pull/3161/files#diff-1087faa40945d18c2dd126f1c39dcfa4R349.  You probably want your change done in that function.